### DISCUSSION TITLE
Reversing fastapi routes

### DISCUSSION THREAD

**Discussion by devxpy** [GooeyAI/gooey-server#374] [Programming Guide]
Whenever you want to refer to a server url in code, reverse route fn to URLs / Paths using `get_route_url()` and `get_path_url()`

Avoid hardcoding the url paths

```py
# bad
ACCOUNT_URL = str(furl(settings.APP_BASE_URL) / "account" / "")

# good
from daras_ai_v2.fastapi_tricks import get_route_url 
from routers.account import account_route

account_url = get_route_url(account_route)
```

**Comment by devxpy** [comment]
PS we use this extensively to build recipe URLs https://github.com/dara-network/ddgai/blob/e7b69bc815eb0ea79dcd9db7bf1262d3b1bb422f/routers/root.py#L729

**Comment by nikochiko** [comment]
do you have something in mind for when this will run into a circular import?

e.g. when we define the routes in a parent `routes/orgs.py` but the actual view logic lives in `daras_ai_v2/orgs.py` or `orgs/views.py`.

a common use case would be to redirect to another route.

**Reply by nikochiko** [reply]
correction: `routers/account.py`, not `routes/orgs.py`

**Reply by devxpy** [reply]
just import locally inside the function

https://github.com/dara-network/ddgai/blob/c971f7afc8be17566489c39c2b85e17a42a30cfd/daras_ai_v2/billing.py#L463
